presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayBlock.java

```diff
@@ -88,7 +88,7 @@ private void calculateSize()
     {
         int valueStart = offsets[arrayOffset];
         int valueEnd = offsets[arrayOffset + positionCount];
-        sizeInBytes = values.getRegion(valueStart, valueEnd - valueStart).getSizeInBytes() + ((Integer.BYTES + Byte.BYTES) * this.positionCount);
+        sizeInBytes = intSaturatedCast(values.getRegion(valueStart, valueEnd - valueStart).getSizeInBytes() + ((Integer.BYTES + Byte.BYTES) * (long) this.positionCount));
```

Comment:
That's a fair question. The fact is that the `Block` interface defines: `int getSizeInBytes();` and I don't know why this assumption (making size in bytes an int) has been made. I handled the potential overflow similar to the way it's handled in other block implementations.